### PR TITLE
Remove dependence on active_support

### PR DESCRIPTION
**Pull request by jason-o-matic**
It wasn't explicit before, and caused an issue with capistrano. Instead
of making the dependency explicit we decided to remove it since that's
where we'd like to head in the future anyway.

In order to test this change we needed a minimal gemfile to ensure other
gems didn't add these methods. This would cause tests to fails because
gems weren't loaded, so the tests have been updated to skip themselves
if their dependencies aren't loaded. This also prints messages which can
help us make sure that the tests that are being skipped should actually
be skipped.

### CODE REVIEW

File: lib/metrician/core_ext.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+          if block_given? && key?(current_key)
+            block.call(current_key, this_value, other_value)
+          else
+            other_value
+          end
+        end
+      end
+
+      self
+    end if !{}.respond_to?(:deep_merge!)
```
Comment (janxious) [inline lib/metrician/core_ext.rb:26]: so funkay. love it.

**Review by janxious — APPROVED** [review]

File: lib/metrician/core_ext.rb (hunk 1 above)
Comment (mediocretes) [reply lib/metrician/core_ext.rb:26]: mother_of_god.jpg

File: lib/metrician/core_ext.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
+begin
+  require 'active_support/core_ext/hash/deep_merge'
+rescue LoadError
+  class Hash
+    # active_support
+    def deep_merge(other_hash, &block)
+      dup.deep_merge!(other_hash, &block)
```
Comment (tonydewan) [inline lib/metrician/core_ext.rb:7]: Where is `dup` defined?

Comment (jason-o-matic) [reply lib/metrician/core_ext.rb:7]: It's the ruby standard dup method from Object: https://ruby-doc.org/core-2.4.1/Object.html#method-i-dup

**Pull request merged**